Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/textfile/TextFileDAO.java

```diff
@@ -0,0 +1,109 @@
+import edu.asu.spring.quadriga.dao.textfile.ITextFileDAO;
+import edu.asu.spring.quadriga.dto.TextFileDTO;
+import edu.asu.spring.quadriga.dto.WorkspaceDTO;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+
+@Repository
+@Transactional
+public class TextFileDAO extends BaseDAO<TextFileDTO> implements ITextFileDAO {
+
+    @Autowired
+    private SessionFactory sessionFactory;
+
+    private static final Logger logger = LoggerFactory.getLogger(TextFileDAO.class);
+
+    @Override
+    public List<TextFileDTO> getTextFileDTObyWsId(String wsId) {
+
+        List<TextFileDTO> tfDTO = null;
+        try {
+            Query query = sessionFactory.getCurrentSession().createQuery(
+                    "from TextFileDTO txtFiles where txtFiles.workspaceId =:wsId");
+            query.setParameter("wsId", wsId);
+            tfDTO = (List<TextFileDTO>)query.list();
+            System.out.println(tfDTO);
+        } catch (HibernateException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+        return tfDTO;
+
+    }
+
+    @Override
+    public List<TextFileDTO> getTextFileDTObyProjId(String projId) {
+
+        List<TextFileDTO> tfDTO = null;
+        try {
+            Query query = sessionFactory.getCurrentSession().createQuery(
+                    "from TextFileDTO txtFiles where txtFiles.projectId =:projID");
+            query.setParameter("projId", projId);
+            tfDTO = (List<TextFileDTO>)query.list();
+            System.out.println(tfDTO);
+        } catch (HibernateException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+        return tfDTO;
+
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * edu.asu.spring.quadriga.dao.textfile.ITextFileDAO#saveTextFileDTO(edu.asu
+     * .spring.quadriga.dto.TextFileDTO) Method to save text file properties in
+     * the db.
+     * 
+     */
+    
+    @Override
+    public boolean saveTextFileDTO(TextFileDTO txtFileDTO) throws QuadrigaStorageException {
+        try {
+            sessionFactory.getCurrentSession().save(txtFileDTO);
+            return true;
+        } catch (Exception e) {
+            throw new QuadrigaStorageException(e);
+        }
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see edu.asu.spring.quadriga.dao.impl.BaseDAO#getDTO(java.lang.String)
+     * Returns a Text DTO for db operations
+     */
+    @Override
+    public TextFileDTO getDTO(String id) {
+        return getDTO(TextFileDTO.class, id);
+    }
+
+    @Override
+    public TextFileDTO getTextFileDTO(String textId) {
+        TextFileDTO tfDTO = null;
+
+         try {
+            tfDTO = (TextFileDTO) sessionFactory.getCurrentSession().get(TextFileDTO.class, textId);
+        } catch (HibernateException e) {
+            logger.error("Retrieve Text File details method :", e);
+        }
+        return tfDTO;
+    }
+

```

Comment:
check the latest version of projectDAO. Your dao should override the getIdPrefix method. Add a prefix (e.g. "TXT") to the projectconstants.properties file and then return that in the overriding method.